src/main/kotlin/com/okp4/connect/cosmos/CosmosSourceTask.kt

```diff
@@ -57,6 +57,10 @@ class CosmosSourceTask : SourceTask() {
                     .map { asSourceRecord(it) }
                     .toList()
             }
+        }.also {
+            it.takeIf { it.isNotEmpty() }?.let { records ->
+                lastBlockHeightFromOffsetStorage = records[records.size - 1].sourceOffset()[HEIGHT_FIELD] as Long
```

Comment:
Could be simplified a bit I guess:

```suggestion
            it.lastOrNull()?.run {
                lastBlockHeightFromOffsetStorage = sourceOffset()[HEIGHT_FIELD] as Long
            }
```

Reply:
Love it! Thank you 😉